src/daemon/plugin.c

```diff
@@ -1078,14 +1078,66 @@ static int plugin_compare_read_func(const void *arg0, const void *arg1) {
     return 0;
 } /* int plugin_compare_read_func */
 
+/*
+ * Set the time of the first read according to the settings AlignRead and
+ * Interval in collectd.conf.
+ *
+ * The AlignRead value is in seconds. Values greater than or equal to 60
+ * determine the minute of an hour (integer part after division with 60).
+ * Values between 0 and 59 (mod 60) determine the second of a minute. Decimal
+ * places set the millisecond of a second.
+ */
+static void plugin_set_next_read_time(read_func_t *rf) {
```

Comment:
Disclaimer: I'm not familiar with collectd's code. But from what I can see, `plugin_set_next_read_time` is only called once when the plugin is registered. To me, the current function name implies that it's called to determine _every_ `next_read_time`, so I'd propose to rename it to more clearly reflect the functionality - maybe `plugin_set_first_read_time`?

Slightly related, only taking `AlignRead` into account once might be a problem if system clock is reset. When synchronizing multiple systems over NTP, they all might have their own time drift and adjust differently at their own intervals. Not sure if that is something worth to consider...

Reply:
I just renamed the function and replaced the "float" with "double". 

You might also be right with the time drift and/or clock reset. I am just thinking about how often doing the realignment. With another config setting, dependent on the AlignRead and/or Interval setting? As the realignment was on the critical path, I would not like to do it with every read.

Reply:
Realignment of reads should happen automatically. Starting with the initial read time (absolute timestamp), which gets aligned with this feature, collectd determines the follow-up reads by adding the interval (or a multiple of the interval if a read took too long). Thus, all "next-reads" are absolute timestamps that are aligned.